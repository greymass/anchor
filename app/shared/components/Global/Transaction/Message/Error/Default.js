import React, { Component } from 'react';
import { withTranslation } from 'react-i18next';
import { Header, Message } from 'semantic-ui-react';
import ReactJson from 'react-json-view';

export class GlobalTransactionMessageErrorDefault extends Component<Props> {
  render() {
    const {
      error,
      t
    } = this.props;

    let errorMessage = error.error && error.error.details && error.error.details[0] && error.error.details[0].message;

    if (errorMessage) {
      const errorMessageArray = errorMessage.split('assertion failure with message:');
      errorMessage = errorMessageArray[1] &&
        errorMessageArray[1]
          .trim()
          .split(' ')
          .slice(0, 8)
          .join('_');
      // The above line will basically take the first 8 words
      // of message and turn it into a locale key.
      errorMessage = errorMessage && errorMessage.replace(/\(|\)|%/g, '');
      errorMessage = errorMessage || errorMessageArray[0];
    }
    return (
      <Message negative>
        <Header>{t('error')}</Header>
        <p key={error}>
          {(errorMessage)
            ? t(errorMessage)
            : `${error.message} (${error.error ? error.error : ''})`
          }
        </p>
        {(typeof error === 'object' && !(error instanceof Error)) ? (
          <ReactJson
            collapsed={1}
            collapseStringsAfterLength={60}
            displayDataTypes={false}
            displayObjectSize={false}
            iconStyle="square"
            name={null}
            src={error}
            style={{ padding: '1em' }}
          />
        ) : ''}
      </Message>
    );
  }
}

export default withTranslation('global')(GlobalTransactionMessageErrorDefault);
