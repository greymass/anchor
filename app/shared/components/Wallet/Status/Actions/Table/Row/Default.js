// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Icon } from 'semantic-ui-react';

class WalletStatusActionsTableRowDefault extends Component<Props> {
  render() {
    const {
      name,
      data,
      t
    } = this.props;

    const iconStyle = {
      margin: '5px',
      marginRight: '20px',
      float: 'left',
      verticalAlign: 'top'
    };

    return (
      <div>
        <Icon
          name="arrow circle up"
          size="large"
          style={iconStyle}
        />
        <div>
          {`${t('actions_table_row_default_header_one')} ${name} ${t('actions_table_row_default_header_two')} ${data.join(' | ')}.`}
        </div>
      </div>
    );
  }
}

export default translate('actions')(WalletStatusActionsTableRowDefault);

