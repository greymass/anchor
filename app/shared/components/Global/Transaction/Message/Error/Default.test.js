import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalTransactionMessageErrorDefault } from './Default';

const mockProps = {
  error: 'error message',
  t: text => text,
};

describe('GlobalTransactionMessageErrorDefault', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalTransactionMessageErrorDefault {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
