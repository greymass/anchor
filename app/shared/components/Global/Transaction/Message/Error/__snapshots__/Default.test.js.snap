// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTransactionMessageErrorDefault renders correctly 1`] = `
<div>
  <p
    key="error message"
  />
</div>
`;
